src/endpoints/borrowed/post/create_borrowed.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from fastapi import Depends, HTTPException, status
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from src.dependencies import get_current_user, get_db
+from src.endpoints.borrowed.router_init import router
+from src.models import all_models
+from src.schemas.borrowed import BorrowedSchema
+
+
+@router.post("/", response_model=None, status_code=status.HTTP_201_CREATED)
+async def create_borrowed(
+    borrowed: BorrowedSchema,
+    user: dict = Depends(get_current_user),
+    db: Session = Depends(get_db),
+) -> BorrowedSchema:
+    """
+    This function will be used to create a new borrowed.
+    Parameters:
+        borrowed: The borrowed data.
+        db: The database session.
+    Returns:
+        borrowed: The created borrowed.
+    """
+    logging.info(f"Creating new borrowed in database with user ID: {borrowed.user_id}")
+    copy = (
+        db.scalars(
+            select(all_models.Copy).where(all_models.Copy.id == borrowed.copy_id)
+        )
+        .unique()
+        .one_or_none()
+    )
+    if copy is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Copy with given ID does not exist",
+        )
+    if copy.status != "available":
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Copy with given ID is not available",
+        )
+    user = (
+        db.scalars(
+            select(all_models.User).where(all_models.User.id == borrowed.user_id)
+        )
+        .unique()
+        .one_or_none()
+    )
+    if user is None:
```

Comment:
I think user_id should not be asked as input. The logged in user's id should be used itself. If we ask id from user then it means one user can borrow for another user, which is wrong.

Reply:
Resolved.